creme/stream/datasets/agrawal.py

```diff
@@ -0,0 +1,316 @@
+import numpy as np
+
+from . import base
+from creme.utils.skmultiflow_utils import check_random_state
+
+
+class AGRAWALGenerator(base.SyntheticDataset):
+    """Agrawal stream generator.
+
+    The generator was introduced by Agrawal et al., and was common source
+    of data for early work on scaling up decision tree learners.
+    The generator produces a stream containing nine features, six numeric and
+    three categorical.
+    There are ten functions defined for generating binary class labels from the
+    features. Presumably these determine whether the loan should be approved.
+    The features and functions are listed in the original paper [1]_.
+
+    | feature name | feature description  | values                                                              |
+    |--------------|----------------------|---------------------------------------------------------------------|
+    | salary       | the salary           | uniformly distributed from 20k to 150k                              |
+    | commission   | the commission       | if (salary < 75k) then 0 else uniformly distributed from 10k to 75k |
+    | age          | the age              | uniformly distributed from 20 to 80                                 |
+    | elevel       | the education level  | uniformly chosen from 0 to 4                                        |
+    | car          | car maker            | uniformly chosen from 1 to 20                                       |
+    | zipcode      | zip code of the town | uniformly chosen from 0 to 8                                        |
+    | hvalue       | value of the house   | uniformly distributed from 50k x zipcode to 100k x zipcode          |
+    | hyears       | years house owned    | uniformly distributed from 1 to 30                                  |
+    | loan         | total loan amount    | uniformly distributed from 0 to 500k                                |
+
+    Parameters
+    ----------
+    classification_function: int (Default=0)
+        Which of the four classification functions to use for the generation.
+        The value can vary from 0 to 9.
+
+    random_state: int, RandomState instance or None, optional (default=None)
```

Comment:
I've been using `seed` as a convention instead of `random_state`. I get that `random_state` is what `sklearn` does, I just find that in most cases users specify an integer and not a `numpy.random.RandomState`.

Also, I've been using the `random` module as a convention instead of `numpy.random`. The reason why is that `random` is usually faster for generating a single value, which happens a lot in a streaming context. Here's a quick benchmark on my laptop:

```py
import random
import numpy as np

seed = 42

rng = random.Random(seed)
np_rng = np.random.RandomState(seed)

%timeit rng.random()
%timeit np_rng.random()
```

```
66.5 ns ± 0.383 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
298 ns ± 2.72 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

Therefore, I really encourage us to use `random` instead of `numpy.random` when possible. For instance, this is what I've done in [`SEA`](https://github.com/creme-ml/creme/blob/the-merge/creme/stream/datasets/sea.py).

Reply:
Agree, in this case using `random` works well. Replaced.

Reply:
I rolled back the change of the random number generator in order to use NumPy's. This is to ensure replicability with the previous implementation when using the same seed values. The variable/attribute name was kept as `seed`.